Doc/c-api/unicode.rst

```diff
@@ -1563,6 +1563,16 @@ object.
    On success, return ``0``.
    On error, set an exception, leave the writer unchanged, and return ``-1``.
 
+.. c:function:: int PyUnicodeWriter_WriteUCS4(PyUnicodeWriter *writer, Py_UCS4 *str, Py_ssize_t size)
+
+   Writer the UCS4 string *str* into *writer*.
+
+   *size* is a number of UCS4 characters. If *size* is equal to ``-1``, get the
+   string length (search the NUL character).
```

Comment:
Is there a use case for null-terminated UCS4 string? `_PyUnicode_FromUCS4()` does not support this, as well as `PyUnicode_FromKindAndData()`. If you use the UCS4 buffer to build a string, you know how much characters were written into it.

Note also that `_PyUnicode_FromUCS4()` is not public. The user is supposed to use more general `PyUnicode_FromKindAndData()`.

Reply:
> Is there a use case for null-terminated UCS4 string?

I don't know. I removed this code: size must be positive.